src/main/java/utility/MarketTracker.java

```diff
@@ -16,38 +18,34 @@
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-/**
- * A singleton class representing the stock market
- */
-
-public class StockMarket {
+public class MarketTracker {
 
     // market information update interval in milliseconds
     private static final long INITIAL_UPDATE_MARKET_INTERVAL = 60000; // initial interval in milliseconds
```

Comment:
Ok here  I can make the comment: move the constants to the resource as a configuration file.